Orig-From: Jan Engelhardt <jengelh@inai.de>
Orig-Date: 2015-02-03 00:43:03.433735818 +0100
From: Paolo Stivanin <info@paolostivanin.com>
Date: 2020-06-17 08:54:21+0000

muparser broke the ABI between 2.2.3 and 2.2.4 by changing.
The abidiff(1) report from libabigail:

  [C]'method std::size_t mu::ParserError::GetPos()' has some indirect sub-type changes:
    return type changed:
      entity changed from typedef std::size_t to int
        name changed from 'long unsigned int' to 'int'
        size changed from 64 to 32 bits
        alignment changed from 64 to 32 bits

Work around this messup by using the full version as unique SONAME.

diff -ru a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2020-06-17 10:52:15.574640873 +0200
+++ b/CMakeLists.txt	2020-06-17 00:09:53.000000000 +0200
@@ -74,7 +74,7 @@
 endif()
 set_target_properties(muparser PROPERTIES
     VERSION ${MUPARSER_VERSION}
+    SOVERSION ${MUPARSER_VERSION}
-    SOVERSION ${MUPARSER_VERSION_MAJOR}
 )
 
 # Install the export set for use with the install-tree
